### ISSUE TITLE
Quick question about ScanNet and DTU data preprocessing

### ISSUE THREAD

**Issue by greeneggsandyaml** [rakhimovv/npbgpp#1]
Dear authors,

Thank you so much for your great work and clean code. I had a quick question -- how do you preprocess your ScanNet and DTU data for pretraining? For ScanNet, I believe this would involve generating `full.ply` and `images/` files for each scan in the dataset. For DTU, I believe this involves generating masks and point clouds. 

Your help is greatly appreciated. 

All the best,

**Comment by TArdelean** [comment]
Hello,

Thank you for your interest in our work.
To obtain the point clouds for ScanNet we largely follow the procedure described in the original NPBG paper ([link](https://arxiv.org/abs/1906.08240v3)). The images are obtained from the RGBD stream by sampling frames with a fixed interval (no less than 1 in 20 frames).
For DTU we use the subset of 15 scenes which have masks, provided by IDR ([link](https://github.com/lioryariv/idr)). The point clouds are obtained in the same way for all datasets except ScanNet, as described in the paper, using PatchmatchNet ([link](https://github.com/FangjinhuaWang/PatchmatchNet)).

**Comment by yizhangphd** [comment]
Dear authors, thank you so much for your great work!
Could you give detailed instructions to help me train or test on Scannet and Nerf synthetic？
For example, how should I organize the data？

**Comment by TArdelean** [comment]
Hello,
All datasets are organized so that each scene is in its own folder. For any scene, there is a directory with the images, with the path specified in the config file, e.g. `images_root: ${datasets.images_subroot}/${datasets.scene_name}/images`. 
For ScanNet we take 100 frames along the RGBD stream, or every 20th image if there are more than 2000 total frames. In the same 'scene folder' we also save the extrinsics, intrinsics, and the point cloud (named `full.ply`). For building the point cloud we follow the procedure from NPBG. For some useful code for extracting information from the `.sens` files see this ([link](https://github.com/ish-vlad/scannet-utils)).
For NeRF-synthetic you just download the official data from here ([link](https://drive.google.com/drive/folders/128yBriW1IG_3NJ5Rp7APSTZsJqdJdfc1)). We use the original `.json` files to read camera parameters, so you only need to construct the point cloud and place it inside the corresponding scene folder. As I said above, for the point clouds we use PatchmatchNet. You can use `prepare_patchmatch_format` function from `SceneBase` class to prepare the data for running PatchmatchNet.

**Comment by rakhimovv** [comment]
Added more example scenes and point clouds to the [cloud](https://disk.yandex.ru/d/-1kx0XUlRHNumQ)

**Comment by yizhangphd** [comment]
Thanks for your help.
